Todo_CLI/todoCLIWithStretch.js

```diff
@@ -0,0 +1,107 @@
+///////////////////////////////////////////////////////////////////////////////////
+//
+//  [Homework] Todo CLI – 3Assignment with Stretch
+//  Write an interactive CLI todo list application using Node's readline and fs modules. 
+//  The following describes what each action does. It would be best to implement 
+//  each action as its own function. 
+//  Auther: Harry Ji 
+//  Date: Sep-18-2021  
+//
+///////////////////////////////////////////////////////////////////////////////////
+const fs = require('fs');
+const readline = require('readline');
+
+const rl = readline.createInterface({
+    input: process.stdin,
+    output: process.stdout,
+});
+  
+console.log('\nWelcome to Todo CLI! \n--------------------');
+let todoList =[]; 
+
+function ensureDirectoryExistence(filePath) {
+    var dirname = path.dirname(filePath);
+    if (fs.existsSync(dirname)) {
+      return true;
+    }
+    ensureDirectoryExistence(dirname);
+    fs.mkdirSync(dirname);
+}
+function write_fie(){
+    fs.writeFile(fileName, JSON.stringify(todoList), err => {
+        if (err) {
+            console.log('Could not write to file');
+            console.error(err);
+            return;
+        }
+        console.log(`List saved to "${fileName}"`);
+        MainMenu();
+    });    
+}
+function MainMenu() {
```

Comment:
A function is an action so should start with a verb and don't start it with a capital letter. It's the convention for class, not function.

Reply:
thanks